### PR TITLE
LS-1237: Add documentation for release 3.1 for Install-upgrade & Operations

### PR DESCRIPTION
**Pull request by nikhil-ctds**
This PR contains the following changes -

- Update Connectors Versions
- Note on migration and braking changes
- Update refs to 2.10 with 3.1v.

### CODE REVIEW

File: modules/install-upgrade/partials/supported-versions-tables.adoc [hunk 1]
```diff
@@ -45,49 +45,45 @@ Support covers only the following Software versions for the Open Source Projects
...
 |*Version*
 |Cassandra enhanced sink
-|v2.7.2, v2.10.x
+|v1.6.11
 |Cloud storage sink
-|v2.7.2, v2.10.x
+|v2.10.0
 |Elasticsearch sink
-|v2.7.2, v2.10.x
+|v2.7.2, v2.10.x, v3.1.x
```
Comment (nicoloboschi) [inline modules/install-upgrade/partials/supported-versions-tables.adoc]: actually we should only keep v3.1.x for builtin connectors

File: modules/install-upgrade/partials/upgrade-2.10.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+* Upgrading to 3.1 Luna Streaming should only be done from 2.10 Luna Streaming.
+
+* Upgrade is fully supported for all the components (connectors included).
+
+* If you're upgrading Pulsar SQL from 2.11 or earlier, you should copy config files from conf/presto to trino/conf.
```
Comment (nicoloboschi) [inline modules/install-upgrade/partials/upgrade-2.10.adoc]: Pulsar SQL is not supported by Luna Streaming, please remove this

File: modules/install-upgrade/partials/upgrade-2.10.adoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+If you're downgrading Pulsar SQL to 2.11 or earlier from newer versions, do the vice-versa.
+
+* Prometheus client version has been changed from 0.5.0 to 0.16.0. Metric type UNTYPED is renamed to UNKNOWN.
+The metrics have been renamed because OpenMetrics's counter name needs a _total suffix.
```
Comment (nicoloboschi) [inline modules/install-upgrade/partials/upgrade-2.10.adoc:7]: Can we add the full list ? they're about 20 so it should be readable

File: modules/install-upgrade/partials/upgrade-2.10.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+E.g `pulsar_expired_token_count` changed to `pulsar_expired_token_total`.
+** Reference PR - https://github.com/apache/pulsar/pull/13785[Bump prometheus client version #13785]
+
+* Downgrade is not supported for Bookies, rest all components support downgrade.
```
Comment (nicoloboschi) [inline modules/install-upgrade/partials/upgrade-2.10.adoc]:
```suggestion
* Downgrade is not supported for Bookies and ZooKeeper. All other components such as Broker, Proxy and Functions Worker can be downgraded at any time.  
```

File: modules/install-upgrade/partials/upgrade-2.10.adoc [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+**  Pulsar 3.1 uses RocksDB 7.x which writes in a format that is not compatible with RocksDB 6.x
```
Comment (nicoloboschi) [inline modules/install-upgrade/partials/upgrade-2.10.adoc:73]: let's keep only the reference issue and remove this line

File: modules/install-upgrade/partials/supported-versions-tables.adoc [hunk 6]
```diff
@@ -6,7 +6,7 @@ Support covers only the following Software versions for Apache Pulsar:
 |*Versions*
 
 |Apache Pulsar
-|2.7.2, 2.10.x
+|2.7.2, 2.10.x, 3.1.x
```
Comment (nicoloboschi) [inline modules/install-upgrade/partials/supported-versions-tables.adoc]: just keep 3.1.x
we don't support AP 2.7 and 2.10

File: modules/install-upgrade/pages/migration.adoc [hunk 7]
```diff
@@ -0,0 +1,3 @@
+= Migration from 2.10 Luna Streaming
```
Comment (nicoloboschi) [inline modules/install-upgrade/pages/migration.adoc]:
```suggestion
= Migration from Luna Streaming 2.10
```

File: modules/install-upgrade/partials/upgrade-2.10.adoc (hunk 5 above)
Comment (nikhil-ctds) [reply modules/install-upgrade/partials/upgrade-2.10.adoc:73]: Reference issue is not present for prometheus metrics, only PRs. I will remove this line anyway.

**Review by srinath-ctds — APPROVED** [review]

**Pull request merged**